protocols/v2/noise-sv2/src/cipher_state.rs

```diff
@@ -80,6 +150,17 @@ where
     }
 }
 
+// An enum representing a generic cipher state.
+//
+// [`GenericCipher`] abstracts over different AEAD cipher implementations, allowing for flexible
+// use of either [`ChaCha20Poly1305`] or [`Aes256Gcm`]. It provides methods for encryption,
+// decryption, and key erasure.
+//
+// A generic cipher that can be either [`ChaCha20Poly1305`] or [`Aes256Gcm`].
+//
+// The [`GenericCipher`] enum allows for flexibility in choosing between different AEAD ciphers.
+// It provides methods for encrypting, decrypting, and securely erasing the encryption key,
+// ensuring that sensitive cryptographic material is handled correctly.
```

Comment:
Once the aes256 gets removed, this enum will be redundant

Reply:
This is being tracked in refactor issue.